DevREADME.md

```diff
@@ -99,4 +99,132 @@ You may encounter the failure of npm install, the error message would say:
+
+### General
+1. Use arrow functions over anonymous function expressions.
+2. Always surround loop and conditional bodies with curly braces.
+3. Open curly braces always go on the same line as whatever necessitates them.
+4. `else` goes on the same line as the closing curly brace.
+5. Use two (2) spaces for indentation.
+6. The `export` keyword should be on its own line.
+7. TypeScript errors/warnings need to be resolved.
+8. Function should always provide type declarations for its return and argument values
+9. Function bodies (including getters/setters) must appear on separate lines from their signature declarations regardless of their simplicity. Empty constructors that exist for clarity are acceptable on a single line
+10. Use a maximum of 140 chars per lines; including leading whitespace
+11. Strings should always use single quotes if they do not contain interpolated values:
+```typescript
+  let foo = 'bar';       // static string
+  let baz = `${5 * 5}`;  // interpolated string
+  let qux = 'These aren\'t the droids you\'re looking for.';  // escaped quotes in string
+```
+
+### Classes
+1. Avoid public properties - use setters/getters.
+2. Do not use the `public` keyword for public members - it is implied by default.
+3. Initialize all private properties to an initial value or `null`, unless they 
+   will be initialized in the constructor. If a basic type is used, do not add 
+   a type specifier (e.g. `private _someValue = 0;`).
+4. Ordering of members within a class should be:
+  - `static`
+  - `public`
+  - `protected`
+  - `private`
+5. Start with getters/setters in each scope, then methods. Protected and private
+   properties go last.
+6. Abstract methods go in the same location as they would in the implemented class.
+
+### Examples
+- Interface declaration:
+
+```typescript
+/**
+ * The base message object which can be sent to a message handler.
+ */
+export
+interface IMessage {
+  /**
+   * The type of the message.
+   * @type {string}
+   */
+  type: string;
+}
+```
+
+- If-Else block:
+
+```typescript
+if (parent) {
+  this._parent = null;
+} else if (this.isAttached) {
+  this.detach(); 
+}
+```
+
+
+- Bit flags:
+
+```typescript
+export
+enum WidgetFlag {
+  /**
+   * The widget is attached to the DOM.
+   */
+  IsAttached = 0x1,
+}
+```
+
+```typescript
+export
+class Widget {
+  /**
+   * Test whether the given widget flag is set.
+   * @param {object} flag WidgetFlag
+   * @return {boolean}
+   */
+  testFlag(flag: WidgetFlag): boolean {
+    return (this._wflags & flag) !== 0;
+  }
+
+  /**
+   * Set the given widget flag.
+   * @param {object} flag WidgetFlag
+   * @returns {void}
+   */
+  setFlag(flag: WidgetFlag): void {
+    this._wflags |= flag;
+  }
+
+  /**
+   * Clear the given widget flag.
+   * @param {object} flag WidgetFlag
+   * @returns {void}
+   */
+  clearFlag(flag: WidgetFlag): void {
+    this._wflags &= ~flag;
+  }
+
+  private _wflags = 0;
+}
+```
+
```

Comment:
I want to added a sections for `Tools`:
```
 format tools: tsfmt
 Lint tool: tslint
```

Reply:
That is a great idea! I definitively had some problems with my extension set when I started. Do you think we should do _Required Tools_, _Suggested Tools_ & _Conflicting Tools_. Each is itemized in an unordered list with a very short description is sufficient or shall we go beyond that with important Settings information?
I think my question clarification would effectively swap the information on either side of your colon and then file them under Required Tools heading.

Reply:
I personally use this tool to format the TS codes: https://github.com/vvakame/typescript-formatter, maybe there is other excellent formatting tools.
Only provide the `Suggested Tools` is enough?